### PR TITLE
Fix friend discovery profile synchronization

### PR DESCRIPTION
**Pull request by pauljsnider**
## Summary

- Move trusted Friends discovery fields fully behind server-owned profile projection logic.
- Populate verified legacy and React-app users from private membership plus Firebase Auth name/photo/email fallbacks.
- Include parent, team-owner, and coach/admin team membership, and resync projections when relevant user or team data changes.
- Add a dry-run-first, project-confirmed backfill for existing `publicUserProfiles` records.

## Root cause

Friends search reads `publicUserProfiles`, but legacy users and coach-only accounts could have a missing or stale projection. The client-derived projection only knew parent links, so it could omit team owner/admin membership and sparse legacy profiles could have no searchable name. That is why expected coaches such as Tim and Brian did not appear even though the social UI itself was working.

## Impact

Verified users are discoverable only when they currently share a projected team, including parents, owners, and coaches. Accepted friendships remain stored independently of later team membership, and friend requests continue to use the deterministic `friendships/{sorted-user-ids}` document so pending state survives reloads. Unverified accounts remain gated by the existing sensitive-write policy.

Existing records require the included backfill after the function triggers are deployed. The script is dry-run by default and apply mode requires an exact project confirmation.

## Verification

- `npm test -- --reporter=dot` — 5,081 passed; 118 skipped.
- `npm test --prefix functions` — 325 passed.
- `npm run app:build` — production build and bundle visualizer verification passed.
- Focused public-profile, backfill, social-service, social-rules, and coverage-map tests — 49 passed; 20 expected skips.
- `node --test functions/test/public-user-profile-projection-core.test.cjs` — 5 passed.
- `node --check functions/index.js`
- `node --check functions/public-user-profile-projection-core.cjs`
- `node --check _migration/backfill-public-user-profiles.js`
- `git diff --check`

Manual production validation used the existing `[REDACTED_EMAIL]` account (no new test user): non-shared Tim/Brian results remained correctly hidden, a shared existing smoke account became discoverable after projection backfill, and the sent friend request remained pending after reload.

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -2759,36 +2760,101 @@ function uniqueNonEmptyStrings(values = []) {
...
+      displayName: authRecord.displayName || null,
+      photoUrl: authRecord.photoURL || null,
+      emailVerified: authRecord.emailVerified === true
+    };
+  } catch (error) {
+    functions.logger.warn('Unable to load Auth identity for public profile projection.', {
+      userId,
+      error: error?.message || String(error)
+    });
+    return {};
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply functions/index.js]: Addressed in 42f55c0e8: only `auth/user-not-found` is handled locally (and removes a stale public projection). Operational Auth errors are logged and rethrown so retry-enabled triggers actually retry.

**Comment by pauljsnider** [comment]
Reviewed current head `42f55c0e8a8d`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `_migration/backfill-public-user-profiles.js`: Ineligible accounts retain stale discoverable profiles.** `syncPublicUserProfileProjectionForUser` returns without deleting `publicUserProfiles/{uid}` when `emailVerified !== true`. The backfill likewise skips both unverified and missing-Auth users without removing existing projections. An account that becomes unverified or loses its Auth record can therefore remain searchable using previously projected team membership, contradicting the stated sensitive-write gate. Delete stale projections for every ineligible state and add focused tests covering runtime sync and backfill cleanup.

**Comment by pauljsnider** [comment]
Addressed the current-head PaulBot finding in 65045318a: runtime projection sync now deletes stale profiles for unverified identities; the backfill deletes stale profiles for missing-Auth and unverified users while surfacing operational Auth failures. Added focused cleanup coverage. Local validation: 5,111 root tests, 343 Functions tests, and app production build passed.

**Comment by pauljsnider** [comment]
Reviewed current head `06bc871ebb1e`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `_migration/backfill-public-user-profiles.js`: Auth/user-orphaned projections can remain discoverable.** Cleanup now occurs when `syncPublicUserProfileProjectionForUser` runs, but no Auth deletion/eligibility-change trigger invokes it; the callable rejects unverified users before reaching cleanup, and unchanged user writes skip synchronization when a projection exists. Additionally, the backfill iterates only `users`, so an existing `publicUserProfiles/{uid}` whose private user document was already deleted is never examined. Enumerate existing projections during backfill, add ongoing Auth-deletion/ineligibility cleanup, and add behavioral tests for deleted Auth accounts, newly unverified accounts, and orphaned public profiles.

- **P2 — `functions/index.js`: Mixed-case admin emails produce unstable staff discovery membership.** `getNotificationRecipientTeamIdsForUser` performs case-sensitive `array-contains` queries using only the raw user email and its lowercase form. If `teams.adminEmails` contains `Coach@Example.com` while Auth supplies `coach@example.com`, runtime resynchronization misses that team and can remove membership that the case-insensitive backfill correctly added. Normalize stored admin-email membership or use a normalized server-owned index, with a focused runtime resync test covering mixed casing.

**Comment by pauljsnider** [comment]
Addressed the current-head PaulBot findings in 9c0141f63:

- Added Auth-deletion cleanup plus a daily eligibility sweep so deleted/newly-unverified/orphaned users cannot retain discoverable public profiles.
- Made unchanged user writes re-check Auth eligibility and made the callable remove an ineligible projection before rejecting the request.
- Added a server-owned UID-based staff membership index and case-insensitive Auth resolution so mixed-case legacy admin emails resolve consistently.
- Extended the safe backfill to reconcile staff indexes and remove orphan public profiles.
- Kept the internal index inaccessible to clients through Firestore rules.

Validation at this exact local head:
- Root unit suite: 5,114 passed, 119 skipped
- Functions suite: 350 passed
- React production build: passed
- Focused friend/profile persistence and rules checks: passed
- No Paul-specific identity was added to automated tests.

The external claim remains until exact-head CI and PaulBot review are green; I will then merge this PR.

**Comment by pauljsnider** [comment]
Reviewed current head `9c0141f6306c`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Auth email changes can leave permanent stale staff discovery access.** `syncPublicUserProfilesForTeamChange` identifies former admins by resolving the old email through Firebase Auth. If that user has since changed their Auth email, the lookup returns not-found, so their existing `publicProfileStaffMemberships` record is never selected for deletion. `getNotificationRecipientTeamIdsForUser` then continually restores the stale team from that index, and there is no Auth-update trigger to reconcile it. The former admin can consequently retain visibility into the team’s discoverable profiles after removal. Reconcile memberships by `teamId` on team changes (rather than relying only on current email resolution), handle Auth email changes, and add focused tests covering an email change followed by admin removal and runtime resynchronization.

**Comment by pauljsnider** [comment]
Addressed the current-head stale-membership finding in 903df38dd. Team staff updates now reconcile the complete server-owned membership set by `teamId`, so an existing UID membership is deleted even when the former admin changed their Auth email and the old address no longer resolves. Every removed indexed UID is then included in public-profile resynchronization.

Added behavioral coverage for an Auth email change followed by admin removal and runtime index reload. Exact local-head validation: 5,114 root tests passed, 351 Functions tests passed, React production build passed, and focused persisted-friendship/profile/rules checks passed.

**Comment by pauljsnider** [comment]
Reviewed current head `903df38dd907`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Auth email changes still leave stale staff discovery membership.** There is no Auth `onUpdate` reconciliation. The Auth delete handler only handles deletion, the scheduled sweep only checks existence/verification, and unchanged user writes return early when a projection exists. Consequently, changing an admin’s Auth email can leave their UID-indexed `publicProfileStaffMemberships` entry and old projected team access indefinitely; a later callable resync also reloads that stale index. Reconcile staff membership and projection on Auth email changes, with focused tests for changing away from an admin email and changing into another team’s admin email without a team write.

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Out-of-order or retried team triggers can restore obsolete discovery access.** Each team-write invocation reconciles the entire UID membership index from that event’s historical `afterTeam` snapshot. Firestore triggers are not ordered, and `failurePolicy: true` permits an older event to retry after a newer edit; that older invocation can therefore recreate removed staff memberships and resync stale team IDs into public profiles. Reconciliation should derive or validate against the current `teams/{teamId}` document at execution time, with a focused stale-event/retry ordering test.

**Comment by pauljsnider** [comment]
Reviewed current head `6350afc758df`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Auth email changes retain stale discovery access for up to 24 hours.** There is still no immediate reconciliation when an admin changes their Firebase Auth email. The new daily sweep eventually repairs the UID-indexed membership and projection, but until then the old `discoveryTeamIds` remains readable and can expose profiles for a team the user no longer administers. Add prompt Auth-identity reconciliation or otherwise prevent stale projected membership from authorizing reads, with a focused email-change test.

- **P2 — Truncated diff (1,602 of 4,102 patch lines omitted): the supplied patch is insufficient for a complete landing-gate review.** This PR changes security rules, Firebase functions, migration behavior, client synchronization, and 114 files, while roughly 39% of the patch is unavailable. The omitted portion prevents exhaustive review for additional authorization/data-boundary regressions and recurring instances. Split the PR or obtain a human review of the complete diff before landing.

**Comment by pauljsnider** [comment]
Reviewed current head `1081eb09218a`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`: Mixed-case `adminEmails` memberships are immediately removed during profile synchronization.** `reconcilePublicProfileStaffMembershipsForAuthUser()` queries `adminEmails` using only the Auth email’s raw/canonical-lowercase forms. Firestore `array-contains` is case-sensitive, so an Auth email such as `coach@example.com` cannot match a stored `Coach@Example.com`. Although the team-write handler initially resolves that address case-insensitively and creates the UID membership, its subsequent `syncPublicUserProfileProjectionForUser()` call runs this per-user reconciliation and deletes it again. The same failure occurs through callable sync, user-write sync, and the scheduled sweep, also undermining the indexed notification fallback. Normalize stored admin emails or derive memberships through a case-insensitive authoritative index, and add an orchestration-level test proving a mixed-case team admin remains discoverable after each synchronization path.

**Comment by pauljsnider** [comment]
Reviewed current head `0943d87ddabf`.

paulbot-current-head-review: blocking findings

- **P1 — `firestore.rules`, `functions/index.js`: Auth email changes still leave stale team-discovery authorization until the old ID token expires.** The rule compares `publicProfileAuthIdentities/{uid}.email` with `request.auth.token.email`, but an already-issued Firebase ID token retains the old email after an admin changes the Auth record. Worse, `syncPublicUserProfileProjection` trusts those stale token claims and can rewrite the identity/projection using the old email instead of loading the current Admin Auth record. The old session therefore remains authorized until token refresh or the scheduled sweep, so the previous immediate-reconciliation finding is not fully fixed. Reconcile against current Admin Auth identity and revoke/otherwise invalidate stale authorization, with a focused test covering an email change while an old token remains active.

- **P2 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: New or email-changed users still cannot acquire an unindexed mixed-case admin membership.** Per-user reconciliation queries `adminEmails` only with the Auth email’s raw and lowercase forms. If a team already stores `Coach@Example.com` and no UID membership was created—commonly because the invite preceded account creation, or the user later changed to that email—Firestore’s case-sensitive `array-contains` cannot find the team, while `loadAuthoritativePublicProfileStaffTeamIds()` can only validate memberships that already exist. This affects callable sync, the user-write trigger, and the scheduled sweep, leaving the coach undiscoverable indefinitely until the team is rewritten or the backfill is rerun. Add case-insensitive authoritative discovery for previously unindexed memberships and test each synchronization path.

**Comment by pauljsnider** [comment]
Reviewed current head `d39ecf21c7d1`.

paulbot-current-head-review: blocking findings

- **P1 — `firestore.rules`, `functions/index.js`: An old ID token retains staff-discovery access after the Auth email changes.** Authorization compares `request.auth.token.email` with `publicProfileAuthIdentities/{uid}.email`. Immediately after an Auth email change, both the already-issued token and stored identity still contain the old email, so the old session remains authorized until another profile synchronization or the daily sweep updates the identity. The added test only checks a *new-email token* against the old identity and does not cover the vulnerable old-token case. Revoke/invalidate stale authorization during email changes or use current server-authoritative identity, and test an already-issued old token.

- **P1 — `functions/public-user-profile-projection-core.cjs`, `firestore.rules`: Client-sourced `users/{uid}.coachOf` is promoted into an authorization-bearing projection without being protected as membership state.** `derivePublicProfileTeamIds()` now includes `userData.coachOf`, while the rules’ protected `userMembershipFields()` list still omits `coachOf`. A user able to update their own profile can therefore claim an arbitrary team ID; the user-write trigger then places it in `publicUserProfiles.discoveryTeamIds`, which `currentUserPublicProfileTeamIds()` trusts for reading other users’ profiles. Remove this source or make it server-controlled and authoritatively validate every claimed coaching membership, with an emulator test proving users cannot self-grant discovery access.

- **P2 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Routine synchronization performs unbounded full-team scans, including redundant scans during the scheduled sweep.** `syncPublicUserProfileOnUserWrite` computes `sourceChanged` but never short-circuits, so even unrelated user activity runs `loadCaseInsensitivePublicProfileStaffTeamIds()` across every team. The scheduled sweep performs that scan twice per eligible profile—once through `reconcileAuthIdentity` and again through `syncEligibleProfile`—and team-write fan-out repeats it for every affected user. This creates O(users × teams) reads and is likely to time out/retry or cause substantial production load. Use the normalized index for ordinary reconciliation and reserve bounded case-insensitive discovery for actual identity changes/backfill; add tests that unchanged writes and each sweep entry avoid duplicate/global scans.

**Comment by pauljsnider** [comment]
Reviewed current head `1fc5fcea26b2`.

paulbot-current-head-review: blocking findings

- **P1 — `firestore.rules`, `functions/index.js`, `tests/unit/app-social-rules.test.js`: Stale ID tokens retain staff-discovery authorization after an Auth email change.** Rules still authorize when the token email matches the stored `publicProfileAuthIdentities` email. There is no Auth-email update trigger, so an old token and old identity continue matching until a callable/user synchronization or daily sweep occurs. The new “denies stale” test actually asserts that the old-token context succeeds. Reconcile or invalidate authorization immediately on Auth identity changes and test that an already-issued old token is denied.

- **P1 — `functions/public-user-profile-projection-core.cjs`, `firestore.rules`: Users can still self-grant discovery access through `coachOf`.** `derivePublicProfileTeamIds()` promotes client-sourced `users/{uid}.coachOf` into the authorization-bearing `discoveryTeamIds`, while `userMembershipFields()` still protects only `parentOf`, `parentTeamIds`, `parentPlayerKeys`, and `playerKeys`. A user can claim an arbitrary team and gain access to profiles associated with it. Remove `coachOf` as a trusted source or make it server-controlled and authoritatively validated, with an emulator test covering attempted self-grant.

- **P2 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Routine synchronization still performs unbounded full-team scans.** `syncPublicUserProfileOnUserWrite` computes `sourceChanged` but always calls projection synchronization, which scans every team even for unrelated user writes. The scheduled sweep scans all teams twice per eligible profile—once in `reconcileAuthIdentity` and again in `syncEligibleProfile`. Team-write fan-out also invokes the same global scan for every affected user. This produces O(users × teams) reads and risks timeouts, retries, and substantial production load. Short-circuit unchanged writes and avoid duplicate/global scans in sweep and team fan-out paths, with focused call-count tests.

**Comment by pauljsnider** [comment]
Reviewed current head `d481c13f3ce5`.

paulbot-current-head-review: blocking findings

- **P1 — `firestore.rules`: Users can self-grant profile-discovery access through `coachOf`.** `currentUserPublicProfileTeamIds()` now directly trusts `users/{uid}.coachOf`, but `userMembershipFields()` still omits `coachOf` from protected membership state. A user can therefore add an arbitrary team ID and read profiles projected for that team. The new denial test does not prove a valid owner-profile update containing only this field is rejected. Make `coachOf` server-controlled and add an emulator test using an otherwise valid existing user document.

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Auth email changes leave former staff membership and notifications active indefinitely.** The daily eligibility sweep calls `syncPublicUserProfileProjectionForUser(..., { useIndexedStaffMemberships: true })`, so it reuses `publicProfileStaffMemberships` without validating them against the user’s current Auth email. There is no Auth-update trigger or sweep reconciliation path. Consequently, after an admin changes email, their old team remains in `discoveryTeamIds`, and `getNotificationRecipientTeamIdsForUser()` continues trusting that stale index for team notifications until an unrelated team write, relevant user write, or callable reconciliation occurs. Reconcile indexed memberships against current Auth identity during the sweep and test the complete email-change lifecycle.

- **P2 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`, `js/db.js`: Routine client synchronization still performs full-team scans.** Every owner `syncPublicUserProfile()` invocation—including normal login/bootstrap flows—calls `syncPublicUserProfileProjection`, which reaches `loadCaseInsensitivePublicProfileStaffTeamIds()` and scans the entire `teams` collection. The same global scan occurs for every relevant `syncPublicUserProfileOnUserWrite` event, including name, photo, and parent-membership edits unrelated to staff identity. This remains production load proportional to routine user activity × all teams. Use the UID index for ordinary projection refreshes and reserve full discovery for actual Auth identity changes or controlled backfill, with focused call-count tests for both callable and user-write paths.

**Comment by pauljsnider** [comment]
Reviewed current head `3696708cacfc`.

paulbot-current-head-review: blocking findings

- **P1 — `firestore.rules`, `tests/unit/app-social-rules.test.js`: `coachOf` still permits self-granted discovery access.** `currentUserPublicProfileTeamIds()` trusts `users/{uid}.coachOf`, while `userMembershipFields()` still excludes `coachOf` from protected server-owned membership fields. The added denial test does not demonstrate rejection of an otherwise valid existing owner-profile update. Protect `coachOf` as server-controlled and add an emulator test using a complete valid user document.

- **P1 — `firestore.rules`: Authoritative owner/admin memberships are projected but not used to authorize their searches.** Server logic places team-owner and `adminEmails` memberships in `publicUserProfiles.discoveryTeamIds`, but rules derive the viewer’s allowed teams only from `parentTeamIds` and `coachOf`. An owner or email-based admin without a legacy `coachOf` entry therefore remains unable to query profiles for their team—the central behavior this PR claims to fix. Authorize from server-owned projected membership and add focused owner/admin tests without `coachOf`.

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Ineligible Auth cleanup leaves staff notification authorization active.** `syncPublicUserProfileProjectionForUser()` removes only the public profile and Auth-identity record when an account becomes missing/unverified, leaving `publicProfileStaffMemberships` intact. The scheduled sweep likewise skips reconciliation for missing users and, for unverified users whose email is unchanged, deletes only the profile. Existing `notificationRecipients` are not resynchronized in either case, so revoked/unverified staff can continue receiving team notifications. Clear indexed memberships and resynchronize every affected team for both missing and unverified identities, with lifecycle tests covering unchanged-email verification revocation and a missed Auth-delete trigger.

**Comment by pauljsnider** [comment]
Reviewed current head `c2e546cf505d`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`, `functions/public-user-profile-lifecycle-core.cjs`: Auth-dependent staff membership remains stale for up to 24 hours.** Both `syncPublicUserProfileProjection` and `syncPublicUserProfileOnUserWrite` use `useIndexedStaffMemberships: true` without reconciling a missing or changed `publicProfileAuthIdentities` record. Consequently: (1) an email-based admin invited before account creation is not discoverable after signup/login until the daily sweep; and (2) an Auth email change leaves former-team discovery authorization and notifications active—and new-team membership absent—until that sweep. The rules test claiming an old token is denied does not change the stale projection and contradicts the implemented authorization path. Reconcile when the current Auth identity differs from the indexed identity, and add lifecycle tests covering pre-provisioned admin signup plus immediate old/new-email authorization and notification changes.

**Comment by pauljsnider** [comment]
Reviewed current head `c1a56ada6f16`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`: Auth email changes and pre-provisioned admin signups still leave staff discovery/notification membership stale for up to 24 hours.** Both `syncPublicUserProfileProjection` and `syncPublicUserProfileOnUserWrite` pass `useIndexedStaffMemberships: true` without comparing or updating `publicProfileAuthIdentities` or reconciling memberships from the current Auth email. Consequently, an invited admin absent when the team trigger ran is not added at signup/login, while an admin whose Auth email changes retains old-team discovery and notifications and misses new-team access until the daily sweep. Reconcile identity mismatches on both routine paths and add focused tests for immediate signup and old/new-email membership and notification changes.

**Comment by pauljsnider** [comment]
Reviewed current head `c70c80c9ae92`.

paulbot-current-head-review: blocking findings

- **P1 — `functions/index.js`: Auth-dependent staff membership remains stale for up to 24 hours.** Both `syncPublicUserProfileProjection` and `syncPublicUserProfileOnUserWrite` still call `syncPublicUserProfileProjectionForUser` with `useIndexedStaffMemberships: true` without comparing the current Firebase Auth email against `publicProfileAuthIdentities` or reconciling memberships. Thus a pre-provisioned email admin is not discoverable immediately after signup, while an Auth email change retains old-team discovery authorization and notification membership and omits new-team access until the scheduled sweep. `getNotificationRecipientTeamIdsForUser` also consumes the stale UID index, so routine notification refreshes can preserve/recreate the former membership. Reconcile identity mismatches on both routine paths and add integration-focused tests covering pre-provisioned admin signup and immediate old/new-email discovery and notification changes.
